jazz_apigee-proxy-aws/index.js

```diff
@@ -0,0 +1,97 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const errorHandlerModule = require("./components/error-handler.js");
+const logger = require("./components/logger.js");
+const AWS = require('aws-sdk');
+
+module.exports.handler = (event, context, callback) => {
+  logger.init(event, context);
+  const errorHandler = errorHandlerModule();
+  AWS.config.region = event.region;
+  const lambda = new AWS.Lambda();
+
+  //Defining the payload object
+  const payload = {
+    operation: event.operation,
+    body: event.body,
+    query: event.queryStringParameters,
+    headers: event.headers,
+    method: event.httpMethod
+  };
+
+  logger.verbose(`payload : ${JSON.stringify(payload)}`);
+  const params = {
+    FunctionName: event.functionName,
+    InvocationType: 'RequestResponse',
+    LogType: 'Tail',
+    Payload: JSON.stringify(payload)
```

Comment:
`payload`  is a JSON. Shouldn't we supply it as-is instead of stringify ing it. 

Ex: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/browser-invoke-lambda-function-example.html#browser-invoke-lambda-function-example-invoking-lambda-function

Reply:
NO it has to be that way. Below is the error

Expected params.Payload to be a string, Buffer, Stream, Blob, or typed array object

Reply:
This is very interesting! We recently merged code to master after identifying this as a bug - https://github.com/tmobile/jazz/commit/a4f828282fd69b8c70cbbfe72fa71578cc9c2733 

Can you check what we are missing here?

Reply:
here - https://docs.aws.amazon.com/lambda/latest/dg/with-userapp-walkthrough-custom-events-invoke.html. The link you gave is talking abt the response payload. But the input payload is stringified json.